utils.py

```diff
@@ -39,103 +51,643 @@ def get_colors(**kwargs):
+                "enseigne1Etablissement": "enseigne_1",
+                "enseigne2Etablissement": "enseigne_2",
+                "enseigne3Etablissement": "enseigne_3",
+                "activitePrincipaleEtablissement": "activite_principale",
+                "indiceRepetitionEtablissement": "indice_repetition",
+                "denominationUsuelleEtablissement": "nom_commercial",
+                "libelleCommuneEtrangerEtablissement": "libelle_commune_etranger",
+                "codePaysEtrangerEtablissement": "code_pays_etranger",
+                "libellePaysEtrangerEtablissement": "libelle_pays_etranger",
+                "libelleCommuneEtranger2Etablissement": "libelle_commune_etranger_2",
+                "codePaysEtranger2Etablissement": "code_pays_etranger_2",
+                "libellePaysEtranger2Etablissement": "libelle_pays_etranger_2",
+            }
         )
-        fill_elastic.execute(dict())
+        df_dep.to_sql("siret", siren_db_conn, if_exists='append', index=False)
+        siren_db_conn.commit()
+        for row in siren_db_cursor.execute(f"""SELECT COUNT() FROM siret"""):
+            logging.info(
+                f"************ {row} records have been added to the unite_legale table!"
+            )
+    del df_dep
+    commit_and_close_conn(siren_db_conn)
 
-    doc_count = fill_elastic.count_docs(dict())
+
+def count_nombre_etablissements():
+    # Connect to database
+    siren_db_conn, siren_db_cursor = connect_to_db()
+    # create a count table
+    siren_db_cursor.execute(f'''DROP TABLE IF EXISTS count_etab''')
+    siren_db_cursor.execute('''CREATE TABLE count_etab (siren VARCHAR(10), count INTEGER)''')
+    # create index
+    siren_db_cursor.execute('''
+                    CREATE UNIQUE INDEX index_count_siren
+                    ON count_etab (siren);
+                    ''')
+    siren_db_cursor.execute(
+        '''INSERT INTO count_etab (siren, count) SELECT siren, count(*) as count FROM siret GROUP BY siren;''')
+    commit_and_close_conn(siren_db_conn)
+
+
+def count_nombre_etablissements_ouverts():
+    siren_db_conn, siren_db_cursor = connect_to_db()
+    siren_db_cursor.execute(f'''DROP TABLE IF EXISTS count_etab_ouvert''')
+    siren_db_cursor.execute(
+        '''CREATE TABLE count_etab_ouvert (siren VARCHAR(10), count INTEGER)''')
+    siren_db_cursor.execute('''
+                    CREATE UNIQUE INDEX index_count_ouvert_siren
+                    ON count_etab_ouvert (siren);
+                    ''')
+    siren_db_cursor.execute(
+        '''INSERT INTO count_etab_ouvert (siren, count) SELECT siren, count(*) as count FROM siret WHERE etat_administratif_etablissement = 'A' GROUP BY siren;''')
+    commit_and_close_conn(siren_db_conn)
+
+
+def add_liste_enseignes():
+    siren_db_conn, siren_db_cursor = connect_to_db()
+    # Add liste d'enseignes for each établissement
+    add_enseigne = f'''ALTER TABLE siret ADD COLUMN enseignes GENERATED ALWAYS AS
```

Comment:
I think this ALTER TABLE is not necessary. We can resolve the creation of the `enseignes` fields at the querying just before indexing into ES.

Reply:
Same approach that we use for dirigeants with json objects and subqueries

Reply:
Tested and implemented!